bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -299,14 +359,50 @@ func decodeInt(b []byte) (int64, int) {
 	copy(bp[8-n:], b[1:n+1])
 
 	var ret int64
-
 	binary.Read(bytes.NewBuffer(bp), binary.BigEndian, &ret)
 
 	if neg {
-		ret -= int64(sizeLimits[n])
+		return ret - int64(sizeLimits[n]), n + 1
+	}
+
+	if ret > 0 {
+		return ret, n + 1
+	}
+
+	return uint64(ret), n + 1
```

Comment:
It might be worth noting with a comment that this is the only case that returns a `uint64` and why it fell through to this case.